src/waypoint.jsx

```diff
@@ -276,19 +281,30 @@ export default class Waypoint extends React.Component {
       return <span ref={this.refElement} style={{ fontSize: 0 }} />;
     }
 
-    const ref = (node) => {
-      this.refElement(node);
-      if (children.ref) {
-        children.ref(node);
-      }
-    };
+    if (typeof children === 'function') {
+      return children(this.refElement);
+    }
```

Comment:
As mentioned earlier, I don't think we need to support this type of child now that we have the `innerRef` prop support.